tools/compatibility-check/contracts_checker.go

```diff
@@ -0,0 +1,137 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2023 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compatibility_check
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+
+	"encoding/csv"
+
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/pretty"
+	"github.com/onflow/cadence/tools/analysis"
+)
+
+const LoadMode = analysis.NeedTypes
+
+type ContractsChecker struct {
```

Comment:
Nice! At first I thought we could maybe just use the linter (`cadence-tools/lint`) for this, but I like that this is a simpler subset of it and can the Cadence dependency can be more easily updated 👌

Reply:
Exactly! It would've been good to use linter without duplicating, but it makes it a lot harder to update the dependencies and all. Also creates an unnecessary circular dependency between repos.